packages/webamp-modern-2/src/skin/AudioPlayer.ts

```diff
@@ -222,6 +286,112 @@ export class AudioPlayer {
     return dispose;
   }
 
+  onVolumeChanged(cb: () => void): () => void {
+    const handler = () => cb();
+    this._audio.addEventListener("volumechange", handler);
+    const dispose = () => {
+      this._audio.removeEventListener("volumechange", handler);
+    };
+    return dispose;
+  }
+
+  //* this only custom listerner ================================
+
+  // execute the callback everytime the label is trigger
+  on(label:string, callback, checkPast = false): ()=>void {
+    this._listeners.has(label) || this._listeners.set(label, []);
+    this._listeners.get(label).push(callback);
+    if (checkPast)
+        this._fCheckPast(label, callback);
+    const dispose = () => {
+      this.off(label, callback);
+    };
+    return dispose;
+  }
+
+  // remove the callback for a label
+  off(label:string, callback) {
+    // if (callback === true) {
+    //     // remove listeners for all callbackfunctions
+    //     this._listeners.delete(label);
+    //     this._onceListeners.delete(label);
+    // } else {
+        // remove listeners only with match callbackfunctions
+        let _off = (inListener) => {
+            let listeners = inListener.get(label);
+            if (listeners) {
+                inListener.set(label, listeners.filter((value) => !(value === callback)));
+            }
+        };
+        _off(this._listeners);
+        _off(this._onceListeners);
+    // }
+}
+
+
+  // help-function for onReady and onceReady
+  // the callbackfunction will execute, 
+  // if the label has already been triggerd with the last called parameters
+  _fCheckPast(label, callback) {
+    if (this._triggerdLabels.has(label)) {
+        callback(this._triggerdLabels.get(label));
+        return true;
+    } else {
+        return false;
+    }
+  }
+
+  // execute the callback onetime the label is trigger
+  once(label:string, callback, checkPast = false) {
+      this._onceListeners.has(label) || this._onceListeners.set(label, []);
+      if (!(checkPast && this._fCheckPast(label, callback))) {
+          // label wurde nocht nicht aufgerufen und 
+          // der callback in _fCheckPast nicht ausgeführt
+          this._onceListeners.get(label).push(callback);
+      }
+  }
+ 
+  // trigger the event with the label 
+  trigger(label:string, ...args: any[]) {
+      let res = false;
+      this._triggerdLabels.set(label, args); // save all triggerd labels for onready and onceready
+      let _trigger = (inListener, label, ...args) => {
```

Comment:
I think you can remove `...args` here and avoid passing it when you call `_trigger` below. The function will just close over the `...args` from the method call.
```suggestion
      let _trigger = (inListener, label) => {
```